### PR TITLE
Create pull requests template

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **Enhancement**

This PR resolves: **[#3496](https://github.com/artsy/eigen/issues/3496)**

### Description

Created the agreed on pull requests template

#trivial

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+The type of this PR is: **Enhancement**
```
Comment (pvinis) [inline docs/pull_request_template.md:1]: Could we keep the `**` but leave it empty?
The way most of us will use it, I'm guessing, is double-click on the words below, copy, and paste it here.

File: docs/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+<!-- bugfix/feature/enhancement.. -->
```
Comment (pvinis) [inline docs/pull_request_template.md:3]:
```suggestion
<!-- Bugfix/Feature/Enhancement.. -->
```

File: docs/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+This PR resolves: **MX-10000**
```
Comment (pvinis) [inline docs/pull_request_template.md:5]:
```suggestion
This PR resolves: **MX-**
```

File: docs/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+### Description
+
+<!-- Implementation description -->
+
+### Test Plan
+
+<!-- if necessary -->
```
Comment (pvinis) [inline docs/pull_request_template.md:13]:
```suggestion
<!-- If necessary -->
```

File: docs/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+### Screenshots
+
+<!-- add screenshots or simulator recordings if applicable -->
```
Comment (pvinis) [inline docs/pull_request_template.md:17]:
```suggestion
<!-- Add screenshots or simulator recordings if applicable -->
```

File: docs/pull_request_template.md (hunk 1 above)
Comment (MounirDhahri) [reply docs/pull_request_template.md:1]: yes, sure 🎉

File: docs/pull_request_template.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+The type of this PR is: **TYPE**
+
+<!-- Bugfix/Feature/Enhancement.. -->
```
Comment (pvinis) [inline docs/pull_request_template.md:3]: Now that this is merged, I can also congratulate you on your usage of two-dot ellipsis! 💜 🤣

**Pull request merged**